### ISSUE TITLE
Cannot access any properties after calling toObject

### ISSUE THREAD

**Issue by lantw44** [typegoose/typegoose#732]
## Versions

- System: linux
- NodeJS: 16.14.0
- Typescript: 4.7.4

- Compiler / Transpiler: tsc
- Typegoose(NPM): 9.9.0
- mongoose: 6.3.9
- mongodb: 5.0.9

## What is the Problem?

TypeScript doesn't allow one to access properties declared on the class after calling `toObject`.
```
index.ts:30:31 - error TS4111: Property 'email' comes from an index signature, so it must be accessed with ['email'].

30   console.log(user.toObject().email);
                                 ~~~~~


Found 1 error in index.ts:30
```

## Code Example

I copied the example from [Mongoose docs](https://mongoosejs.com/docs/typescript.html) and manually converted it to Typegoose.

tsconfig.json
```json
{
  "compilerOptions": {
    "target": "es2020",
    "module": "commonjs",
    "esModuleInterop": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "noUncheckedIndexedAccess": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitOverride": true
  }
}
```

index.ts
```ts
import { connect } from 'mongoose';
import { prop, getModelForClass } from '@typegoose/typegoose';

class UserClass {
  @prop({ required: true })
  public name!: string;

  @prop({ required: true })
  public email!: string;

  @prop()
  public avatar?: string;
}

const UserModel = getModelForClass(UserClass);

run().catch(err => console.log(err));

async function run() {
  await connect('mongodb://localhost:27017/test');

  const user = await UserModel.create({
    name: 'Bill',
    email: 'bill@initech.com',
    avatar: 'https://i.imgur.com/dM7Thhn.png'
  });
  await user.save();

  console.log(user.email);
  console.log(user.toObject().email);
}
```

## Do you know *why* it happens?

No, but it seems to be related to changes introduced in Mongoose 6.3.2. Downgrading to Mongoose 6.3.1 (with `skipLibCheck: true`) fixed compilation, but Typegoose refused to run because Mongoose was too old.

While removing `noPropertyAccessFromIndexSignature` from tsconfig.json could also fix compilation, it caused the `email` property to have `any` type, which was wrong because I declared it as `string` in the class.

**Comment by lantw44** [comment]
Mongoose does have types for `toObject`, but it doesn't work with Typegoose. It works with the [Mongoose example](https://mongoosejs.com/docs/typescript.html) though.

> * removing `node_modules` folder and re-installing all packages

The `node_modules` folder was installed just before opening the issue.

> * verify that typescript is of at least 4.4 (and if that does not work, try latest)

TypeScript 4.7.4 is already the latest version.

> * verify that `@types/mongoose` is not installed

```
$ npm ls -a
t@1.0.0 /path/to/t
├─┬ @typegoose/typegoose@9.9.0
│ ├── lodash@4.17.21
│ ├── loglevel@1.8.0
│ ├── mongoose@6.3.9 deduped
│ ├── reflect-metadata@0.1.13
│ ├─┬ semver@7.3.7
│ │ └─┬ lru-cache@6.0.0
│ │   └── yallist@4.0.0
│ └── tslib@2.4.0
├─┬ mongoose@6.3.9
│ ├─┬ bson@4.6.4
│ │ └─┬ buffer@5.7.1
│ │   ├── base64-js@1.5.1
│ │   └── ieee754@1.2.1
│ ├── kareem@2.3.5
│ ├─┬ mongodb@4.5.0
│ │ ├── bson@4.6.4 deduped
│ │ ├── denque@2.0.1
│ │ ├─┬ mongodb-connection-string-url@2.5.2
│ │ │ ├─┬ @types/whatwg-url@8.2.2
│ │ │ │ ├── @types/node@18.0.0
│ │ │ │ └── @types/webidl-conversions@6.1.1
│ │ │ └─┬ whatwg-url@11.0.0
│ │ │   ├─┬ tr46@3.0.0
│ │ │   │ └── punycode@2.1.1
│ │ │   └── webidl-conversions@7.0.0
│ │ ├─┬ saslprep@1.0.3
│ │ │ └─┬ sparse-bitfield@3.0.3
│ │ │   └── memory-pager@1.5.0
│ │ └─┬ socks@2.6.2
│ │   ├── ip@1.1.8
│ │   └── smart-buffer@4.2.0
│ ├── mpath@0.9.0
│ ├─┬ mquery@4.0.3
│ │ └─┬ debug@4.3.4
│ │   └── ms@2.1.2
│ ├── ms@2.1.3
│ └── sift@16.0.0
└── typescript@4.7.4
```

**Comment by hasezoey** [comment]
can confirm this is a typegoose problem by not passing it into the document generic, will try to fix (though it will only probably be for 9.10.0)
